internal/k8s/helpers.go

```diff
@@ -83,3 +89,77 @@ func isStatusEqual(objA, objB interface{}) bool {
 	}
 	return false
 }
+
+// IsObjectEqual checks if objects received during update are equal.
+//
+// Make an attempt to avoid comparing full objects since it can be very CPU intensive.
+// Prefer comparing Generation when only interested in spec changes.
+func IsObjectEqual(old, new interface{}) (bool, error) {
+
+	// Fast path for any object: when ResourceVersions are equal, the objects are equal.
+	// NOTE: This optimizes the case when controller-runtime executes full sync and sends updates for all objects.
+	if isResourceVersionEqual(old, new) {
+		return true, nil
+	}
+
+	switch old := old.(type) {
+
+	// Fast path for objects that implement Generation and where only spec changes matter.
+	// Generation is implemented in CRDs, Ingress and IngressClass.
+	case *contour_api_v1alpha1.ExtensionService:
+		return isGenerationEqual(old, new), nil
+	case *contour_api_v1.TLSCertificateDelegation:
+		return isGenerationEqual(old, new), nil
+	case *contour_api_v1alpha1.ContourConfiguration:
+		return isGenerationEqual(old, new), nil
+
+	case *gatewayapi_v1beta1.GatewayClass:
+		return isGenerationEqual(old, new), nil
+	case *gatewayapi_v1beta1.Gateway:
+		return isGenerationEqual(old, new), nil
+	case *gatewayapi_v1beta1.HTTPRoute:
+		return isGenerationEqual(old, new), nil
+	case *gatewayapi_v1alpha2.TLSRoute:
+		return isGenerationEqual(old, new), nil
```

Comment:
Need to add `*gatewayapi_v1beta1.ReferenceGrant` here as well

Reply:
Fixed!